quickjs.h

```diff
@@ -38,14 +38,34 @@ extern "C" {
 
 #if defined(__GNUC__) || defined(__clang__)
 #define js_force_inline       inline __attribute__((always_inline))
-#define __js_printf_like(f, a)   __attribute__((format(printf, f, a)))
 #define JS_EXTERN __attribute__((visibility("default")))
 #else
 #define js_force_inline  inline
-#define __js_printf_like(a, b)
 #define JS_EXTERN /* nothing */
 #endif
 
+/* Borrowed from Folly */
+#ifndef JS_PRINTF_FORMAT
+#ifdef _MSC_VER
+#ifdef _USE_ATTRIBUTES_FOR_SAL
+#undef _USE_ATTRIBUTES_FOR_SAL
+#endif
+#define _USE_ATTRIBUTES_FOR_SAL 1
```

Comment:
Is it safe to do this in a public header file?

Reply:
I'll dig a bit, I'm not sure...

Reply:
I loked in that Windows header file and it already behaves correctly so I dropped the logic there. Looks like that might have been necessary in 2015, but not now.